ts/Extensions/Annotations/Types/TimeCycles.ts

```diff
@@ -0,0 +1,320 @@
+                point.yAxis = options.yAxis;
+            });
+        }
+
+        if (defined(options.xAxis)) {
+            (options.points as MockPointOptions[]).forEach((point): void => {
+                point.xAxis = options.xAxis;
+            });
+        }
+        super.init.call(this, annotation, options, index);
+
+    }
+
+    public setPath(): void {
+
+        this.shapes[0].options.d = this.getPath();
+    }
+    public getPath(): SVGPath {
+
+        return ([getStartingPath(this.startX, this.y)] as SVGPath).concat(getCirclePath(
+            this.pixelInterval,
+            this.numberOfCircles,
+            this.startX,
+            this.y
+        ));
+    }
+
+    public addShapes(): void {
+        const typeOptions = this.options.typeOptions;
+        this.setPathProperties();
+        const shape = this.initShape(
+            merge(typeOptions.line, {
+                type: 'path',
+                d: this.getPath(),
+                points: this.options.points
+            }),
+            0
+        );
+
+        typeOptions.line = shape.options;
+    }
+
+    public addControlPoints(): void {
+        const options = this.options,
+            typeOptions = options.typeOptions as TimeCycles.TypeOptions;
+        options.controlPointOptions.style.cursor = this.chart.inverted ? 'ns-resize' : 'ew-resize';
+
+        typeOptions.controlPointOptions.forEach(
+            (option: Highcharts.AnnotationControlPointOptionsObject): void => {
+                const controlPointsOptions = merge(options.controlPointOptions, option);
+                const controlPoint = new ControlPoint(this.chart, this, controlPointsOptions, 0);
+                this.controlPoints.push(controlPoint);
+            });
+    }
+
+    public setPathProperties(): void {
+        const options = this.options.typeOptions,
+            points = options.points;
+
+        if (!points) {
+            return;
+        }
+        const point1 = points[0],
+            point2 = points[1],
+            xAxisNumber = options.xAxis || 0,
+            yAxisNumber = options.yAxis || 0,
+            xAxis = this.chart.xAxis[xAxisNumber],
+            yAxis = this.chart.yAxis[yAxisNumber],
+            xValue1 = point1.x,
+            yValue = point1.y,
+            xValue2 = point2.x;
+
+        if (!xValue1 || !xValue2) {
+            return;
+        }
+
+        const y = isNumber(yValue) && !isNaN(yValue) ? yAxis.toPixels(yValue) : yAxis.top + yAxis.height,
+            x = isNumber(xValue1) && !isNaN(xValue1) ? xAxis.toPixels(xValue1) : xAxis.left,
+            x2 = isNumber(xValue2) && !isNaN(xValue2) ? xAxis.toPixels(xValue2) : xAxis.left + 30,
+            xAxisLength = xAxis.len,
+            pixelInterval = Math.round(Math.max(Math.abs(x2 - x), 2)),
+            numberOfCircles = Math.floor(xAxisLength / pixelInterval) + 2,
+            pixelShift = (Math.floor((x - xAxis.left) / pixelInterval) + 1) * pixelInterval;

```

Comment:
Could you add some comments here explaining these variables and `+ 2` and `+ 1` operations?